File: hard/12_suffix_trie_construction/suffix_trie_construction.test.js
const  SuffixTrie = require('./suffix_trie_construction');

test("Test Case #1", () => {
  const trie = new SuffixTrie('babc');
  const expected = {
    c: {'*': true},
    b: {
      c: {'*': true},
      a: {b: {c: {'*': true}}},
    },
    a: {b: {c: {'*': true}}},
  };
  expect(trie.root).toEqual(expected);
  expect(trie.contains('abc')).toEqual(true);
});

File: hard/12_suffix_trie_construction/suffix_trie_construction_test.py
import unittest
from suffix_trie_construction import SuffixTrie

class TestProgram(unittest.TestCase):
    def test_case_1(self):
        trie = SuffixTrie("babc")
        # self.assertEqual(trie.root, expected)
        self.assertEqual(trie.contains("abc"), True)
